lib/Applify.pm

```diff
@@ -344,6 +375,28 @@ sub _sub {
   *$fqn = SUB_NAME_IS_AVAILABLE ? Sub::Name::subname($fqn, $code) : $code;
 }
 
+sub _subcommand_activate {
+  my ($self, $name) = @_;
+  return undef unless $name and $name =~ /^\w+/;
+  return undef unless $self->{subcommands}{$name};
+  $self->{subcommand} = $name;
+  {
+    no warnings 'redefine';
+    local *Applify::app = sub {
+      warn "Avoided deep recursion. Do not call app {} from subcommand block!\n";
+    };
```

Comment:
When does this happen? Is it just to help the author of the application? If so, do this instead:

```
local *Applify::app = sub { Carp::confess("Looks like you have a typo in your script! Cannot have app{} inside a subcommand options block.") };
```

Reply:
Pretty much. `confess` `die`s that ok?

Reply:
Could be useful with confess, so you can see in which subcommand definition that you have the bug.